### PR TITLE
Headers and URLs

### PR DESCRIPTION
**Pull request by shubham4443**
1. Added support for traffic splitting based on headers and urls.
2. Style of collapse component inside a form was globally modified specifically for advanced settings. Added a class name for such components so they don't override other collapses.

### CODE REVIEW

**Comment by Saif-Ali-Kamal** [comment]
- matchers payload is not received in request and not stored in redux
- in matchers headers if type is check presence there should be only one field i.e. key

**Comment by Saif-Ali-Kamal** [comment]
- matchers are not visible at time of edit, maybe there is issue in initialValue
- also test it by putting matcher in mirage data

**Review by Saif-Ali-Kamal — APPROVED** [review]

File: src/components/deployments/routing-modal/RoutingModal.jsx [hunk 1]
```diff
@@ -90,6 +92,112 @@ const RoutingRule = props => {
...
+                                          <Form.Item
+                                            {...fieldHeaders}
+                                            name={[fieldHeaders.name, 'type']}
+                                            fieldKey={[fieldHeaders.name, 'type']}
+                                          >
+                                            <Select placeholder="Type" style={{ width: 300 }}>
+                                              <Select.Option value="exact">Exact</Select.Option>
+                                              <Select.Option value="prefix">Prefix</Select.Option>
+                                              <Select.Option value="regex">Regex</Select.Option>
+                                              <Select.Option value="checkPresence">Check presence</Select.Option>
```
Comment (Saif-Ali-Kamal) [inline src/components/deployments/routing-modal/RoutingModal.jsx:156]: change checkPresence to check-presence

**Review by Saif-Ali-Kamal — CHANGES_REQUESTED** [review]
- matchers is optional but if in url and headers if one of the value is filled make other value in respective section mandatory
- try to solve warning that are present in console

**Review by sharadregoti — APPROVED** [review]

**Pull request merged**